lib/puppet/provider/cisco_vpc_domain/cisco.rb

```diff
@@ -51,22 +51,26 @@
     :system_mac,
     :system_priority,
   ]
-  VPC_BOOL_PROPS = [
+  VPC_BOOL_PROPS1 = [
     :auto_recovery,
+    :peer_gateway,
+  ]
```

Comment:
Please add a comment to explain why two bool arrays are needed to ensure the setter processing order.